internal/replacements/replacements_test.go

```diff
@@ -290,13 +290,83 @@ var _ = Describe("replace_misconfigured_pods", func() {
 				pod.Spec.NodeSelector = map[string]string{
 					"dummy": "test",
 				}
-				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log)
+				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
 				Expect(needsRemoval).To(BeFalse())
 				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
-		Context("when UpdatePodsByReplacement is not set and the PodSpecHash doesn't match", func() {
+		When("the securityContext doesn't match", func() {
+			var originalSecurityContext *corev1.PodSecurityContext
+			var originalSpecHash string
+
+			BeforeEach(func() {
+				pClass = fdbv1beta2.ProcessClassStorage
+				remove = false
+				originalSecurityContext = pod.Spec.SecurityContext
+				originalSpecHash = pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey]
+			})
+
+			When("the last applied spec hash is different from desired spec hash", func() {
+				BeforeEach(func() {
+					pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey] = "banana"
+				})
+
+				When("FSGroup is changed", func() {
+					BeforeEach(func() {
+						pod.Spec.SecurityContext = &corev1.PodSecurityContext{FSGroup: pointer.Int64(1234)}
+					})
+
+					When("replaceOnSecurityContextChange is true", func() {
+						It("should need a removal", func() {
+							needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
+							Expect(needsRemoval).To(BeTrue())
+							Expect(err).NotTo(HaveOccurred())
+						})
+					})
+
+					When("replaceOnSecurityContextChange is false", func() {
+						It("should *not* need a removal", func() {
+							needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, false)
+							Expect(needsRemoval).To(BeFalse())
+							Expect(err).NotTo(HaveOccurred())
+						})
+					})
+
+					AfterAll(func() {
```

Comment:
```suggestion
					AfterEach(func() {
```

AfterAll can only be used in an ordered setup: https://onsi.github.io/ginkgo/#setup-in-ordered-containers-beforeall-and-afterall, you have to use `AfterEach`.